brush-core/src/builtins/mapfile.rs

```diff
@@ -12,52 +12,43 @@ pub(crate) struct MapFileCommand {
     delimiter: String,
 
     /// Maximum number of entries to read (0 means no limit).
-    #[arg(short = 'n', default_value = "0")]
+    #[arg(short = 'n', default_value_t = 0)]
     max_count: i64,
 
     /// Index into array at which to start assignment.
     #[arg(short = 'O')]
     origin: Option<i64>,
 
     /// Number of initial entries to skip.
-    #[arg(short = 's', default_value = "0")]
+    #[arg(short = 's', default_value_t = 0)]
     skip_count: i64,
 
     /// Whether or not to remove the delimiter from each read line.
     #[arg(short = 't')]
     remove_delimiter: bool,
 
     /// File descriptor to read from (defaults to stdin).
-    #[arg(short = 'u', default_value = "0")]
+    #[arg(short = 'u', default_value_t = 0)]
     fd: u32,
 
     /// Name of function to call for each group of lines.
     #[arg(short = 'C')]
     callback: Option<String>,
 
     /// Number of lines to pass the callback for each group.
-    #[arg(short = 'c', default_value = "5000")]
+    #[arg(short = 'c', default_value_t = 5000)]
     callback_group_size: i64,
 
     /// Name of array to read into.
+    #[arg(default_value = "MAPFILE")]
     array_var_name: String,
 }
 
 impl builtins::Command for MapFileCommand {
     async fn execute(
         &self,
-        context: commands::ExecutionContext<'_>,
+        mut context: commands::ExecutionContext<'_>,
     ) -> Result<crate::builtins::ExitCode, error::Error> {
-        if self.delimiter != "\n" {
```

Comment:
Nice! You should update the PR title to reflect that you've also added support for other options too (e.g., `-d`, `-s`).